core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -426,7 +427,7 @@ public void loadDependenciesOf(Class<?> clazz) {
             throw new IllegalStateException("Service dependency declared by " + clazz.getName() +
                 " is not a Service: " + dependency.getName());
           }
-          dependencies.add(dependency);
+          dependencies.add((Class<? extends Service>) dependency);
```

Comment:
Had a quick run through the product code (don't really care about the tests as much) - only problem I could see was that this cast could be eliminated if we type the for loop variable properly.